### ISSUE TITLE
New tool: assign-nobles

### ISSUE THREAD

**Issue by myk002** [DFHack/dfhack#3517]
Allows automation of assignment of nobles according to relevant attributes.

Or assigning a specific unit

Overlay on unit sheet allowing assignment of the current unit to noble roles

Autostart option for control-panel for auto assigning starting nobles

**Comment by DragonEyeNinja** [comment]
It might also be prudent to have certain zones designated for nobles that can change around (such as the mayor) to be automatically re-assigned if the current occupant of the position changes.

**Comment by myk002** [comment]
Yeah, this tool would be a good place to stick that functionality. It used to be part of `buildingplan` for some reason, but I didn't port that part
